send-emails-from-a-service/main.bal

```diff
@@ -0,0 +1,164 @@
+configurable string hospitalServicesBackend = "http://localhost:9090";
+final http:Client hospitalServicesEP = check new(hospitalServicesBackend);
+
+service /healthcare on new http:Listener(port) {
+
+    function init() returns error? {
+        check startSendWithOptionsSmtpServer();
+    }
+        
+    resource function post categories/[string category]/reserve(@http:Payload ReservationRequest reservationRequest) 
+        returns http:InternalServerError|http:NotFound|error? {
+
+        email:SmtpConfiguration config = {
+        port: 3025,
+        security: email:START_TLS_NEVER
+    };
+        email:SmtpClient smtpClient = check new("127.0.0.1", "hascode", "abcdef123", config);
+
+        ReservationRequest {
+            patient: {cardNo, ...patient},
+            doctor, 
+            hospital,
+            hospital_id,
+            appointment_date
+        } = reservationRequest;
+
+
+        Appointment|http:ClientError appointment =
+                hospitalServicesEP->/[hospital_id]/categories/[category]/reserve.post({
+            patient,
+            doctor,
+            hospital,
+            appointment_date
+        });
+
+        if appointment !is Appointment {
+            log:printError("Appointment reservation failed", appointment);
+            if appointment is http:ClientRequestError {
+                return <http:NotFound> {body: string `unknown hospital, doctor, or category`};
+            }
+            return <http:InternalServerError> {body: appointment.message()};
+        }
+
+        PaymentSettlement paymentSettlement = {
+            appointmentNumber: appointment.appointmentNumber,
+            doctor: appointment.doctor,
+            patient: appointment.patient,
+            fee: appointment.doctor.fee,
+            confirmed: appointment.confirmed,
+            card_number: cardNo
+        };
+
+        Payment|http:ClientError payment =
+                hospitalServicesEP->/healthcare/payments.post(paymentSettlement);
+
+        if payment !is Payment {
+            log:printError("Payment settlement failed", payment);
+            if payment is http:ClientRequestError {
+                return <http:NotFound> {body: string `unknown hospital, patient, or category`};
+            }
+            return <http:InternalServerError> {body: payment.message()};
+        }
+
+        ReservationResponse reservationResponse = {
+            appointmentNo: appointment.appointmentNumber,
+            doctorName: reservationRequest.doctor,
+            patient: reservationRequest.patient.name,
+            actualFee: payment.actualFee,
+            discount: payment.discount,
+            discounted: payment.discounted,
+            paymentID: payment.paymentID,
+            status: payment.status
+        };
+
+        email:Message email = {
+            to: reservationRequest.patient.email,
+            subject: "Payment Status",
+            body: reservationResponse.toString()
```

Comment:
I know the WSO2 sample has a map and uses the toString representation as the body. But shall we just go with a formatted string directly instead of `ReservationResponse`?